File: proyectSaludVital_Angular/src/app/DTO/obtener-medico-dto.ts
export interface HorarioDTO {
  dia: string;
  horaInicio: string;
  horaFin: string;
}

export interface ObtenerMedicoDTO {
  id: number;
  nombre: string;
  apellido: string;
  numeroLicencia: string;
  telefono: string;
  email: string;
  especialidad: string;
  estado?: string;
  disponible?: boolean;
  tarifaConsulta: number;
  usuarioId: number;
  imagenUrl?: string;
  horarios: HorarioDTO[];
}

File: ProyectoSaludVital-Backend/src/main/java/com/example/practica/DTO/ObtenerMedicoDTO.java
package com.example.practica.DTO;

import java.math.BigDecimal;
import java.util.List;

import com.example.practica.Enum.Especialidad;
import com.example.practica.Enum.EstadoDoctor;


import lombok.*;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class ObtenerMedicoDTO {
    private Long id;
    private String nombre;
    private String apellido;
    private String numeroLicencia;
    private String telefono;
    private String email;
    private Especialidad especialidad;
    private EstadoDoctor estado;
    private boolean disponible;
    private BigDecimal tarifaConsulta;
    private long usuarioId;
    private String imagenUrl;
    private List<HorarioDTO> horarios;
}